analysis/import/SCMLogParser/src/main/kotlin/de/maibornwolff/codecharta/importer/scmlogparser/parser/CommitCollector.kt

```diff
@@ -10,70 +11,97 @@ import java.util.stream.Collector
+                            missingVersionControlledFile // and add it to the list
+                        missingVersionControlledFile.registerCommit(commit, it)
+                    } else {
+                        versionControlledFiles[possibleConflictName]?.registerCommit(commit, it)
+                    }
+                }
+                Modification.Type.DELETE -> {
+                    val filename = renamesMap[possibleConflictName]
+                    versionControlledFiles.remove(if (filename == null) possibleConflictName else filename) //remove a deleted file from list
+                    renamesMap.remove(possibleConflictName) // and remove its references in renames
+                }
+                Modification.Type.RENAME -> {
 
-    private fun addYetUnknownFile(
-        versionControlledFiles: MutableList<VersionControlledFile>,
-        filenameOfYetUnversionedFile: String
-    ): Boolean {
-        val missingVersionControlledFile = VersionControlledFile(filenameOfYetUnversionedFile, metricsFactory)
-        return versionControlledFiles.add(missingVersionControlledFile)
-    }
+                    val renameName: String? = renamesMap[possibleConflictName] //current
+                    val tmp: String? =
+                        if (renameName == null) possibleConflictName else renameName //do we have an entry?
+                    var newVCFFileName = it.currentFilename
+
+                    if (versionControlledFiles.containsKey(it.currentFilename)) {//is the filename contained in vCF
+                        val marker: Int? =
+                            nameConflictsMap[it.currentFilename] //check if the file is already contained with a marker
+                        val newMarker: Int =
+                            if (marker != null) marker + 1 else 0 //increment the marker for additional tracking
 
-    private fun addCommitMetadataToMatchingVersionControlledFiles(
-        commit: Commit,
-        versionControlledFiles: List<VersionControlledFile>
-    ) {
-        commit.filenames.mapNotNull { findVersionControlledFileByFilename(versionControlledFiles, it) }
-                .forEach { it.registerCommit(commit) }
+                        newVCFFileName =
+                            it.currentFilename + "_\\0_" + newMarker // generate new vCF entry we work with it.currentFilename to preserve the string structure
+                    }
+                    if (renameName != null) {
+                        renamesMap.remove(possibleConflictName) //remove old entry
+                        renamesMap[newVCFFileName] = renameName // insert new entry, tracking the current name
+                    } else {
+                        renamesMap[newVCFFileName] = it.oldFilename //add new entry
+                    }
+                    versionControlledFiles[tmp]?.registerCommit(commit, it)
+                }
+
+                else -> versionControlledFiles[possibleConflictName]?.registerCommit(commit, it)
```

Comment:
```suggestion
                else -> versionControlledFiles[VCFName]!!.registerCommit(commit, it)
```

Reply:
Done it, but the issue still persists
Rename is not working correctly
Files that have been renamed persist in CC, and new names are not recognized.